### PR TITLE
Add webp support to texture loader

### PR DESCRIPTION
**Pull request by hanabi1224**
# Objective

- This PR adds webp image format support to TextureLoader behind an optional feature ```webp```

## Solution

- Since webp support in image crate is very limited (e.g. lack of ALPH support), it uses FFI bindings to libwebp instead.

### CODE REVIEW

**Comment by DJMcNab** [comment]
I think there are places where we document our features

We need to do this for this one (we should also mention that it uses native libraries, so will be more difficult to build for someone)

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +92,7 @@ current changes on git with [previous release tags][git_tag_comparison].
 - [Added `set_minimized` and `set_position` to `Window`][1292]
 - [Example for 2D Frustum Culling][1503]
 - [Add remove resource to commands][1478]
+- [Add webp as a supported texture format (C compiler required)][2600]
```
Comment (MinerSebas) [inline CHANGELOG.md:95]: Thats the Changelog of the already released 0.5
0.6 does not have a Changelog yet.

Comment (hanabi1224) [reply CHANGELOG.md]: fixed.

**Comment by mockersf** [comment]
Didn't find an easy answer with a quick read, is the link to libwebp static or dynamic?
* if dynamic, it will need player to have libwebp built on their system for this feature
* if static, released Bevy game with this feature will need to include https://chromium.googlesource.com/webm/libwebp/+/refs/heads/main/COPYING which is not obvious or easy to find

**Comment by hanabi1224** [comment]
@mockersf It's statically linked, regarding license, could you please point me to where I should put up a notice, code doc or feature doc?

**Comment by mockersf** [comment]
we don't have a good place yet for that, I guess for now mentioning this in #1885 when it gets merged should be enough

**Review by mockersf — APPROVED** [review]

**Comment by inodentry** [comment]
So, this PR is stale ... but here is what could happen given the status in current bevy.

From looking at the source code, I see that we are enumerating WebP (incl many others) in `enum ImageFormat`: https://github.com/bevyengine/bevy/blob/main/crates/bevy_render/src/texture/image.rs#L31

However, there is no actual asset loader enablement for it (and a few of the other formats listed there btw). 

It is lacking here: https://github.com/bevyengine/bevy/blob/main/crates/bevy_render/src/texture/image_texture_loader.rs#L20
and here: https://github.com/bevyengine/bevy/blob/main/crates/bevy_render/src/texture/mod.rs#L39

For anyone wanting to pick this up, you could look at the places linked above, as well as the code that was in this PR, for reference. It seems like it should be relatively straighforward to bring this up to date to current bevy.

**Comment by james7132** [comment]
This has been superceded by #8220.

**Pull request closed without merging**